omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/PasswordResetController2_2.java

```diff
@@ -12,59 +12,47 @@
 import java.util.Map;
 
 import org.openmrs.User;
-import org.openmrs.api.InvalidActivationKeyException;
 import org.openmrs.api.UserService;
 import org.openmrs.module.webservices.rest.web.RestConstants;
 import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
 @Controller
-@RequestMapping(value = "/rest/" + RestConstants.VERSION_1)
-public class PasswordResetController extends BaseRestController {
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/passwordreset")
+public class PasswordResetController2_2 extends BaseRestController {
 	
 	@Qualifier("userService")
 	@Autowired
 	private UserService userService;
 	
-	@RequestMapping(value = "/requestpasswordreset", method = RequestMethod.POST)
+	@RequestMapping(value = "/{usernameOrEmail}", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
-	public void requestPasswordReset(@RequestBody Map<String, String> body) {
+	public void requestPasswordReset(@PathVariable("usernameOrEmail") String usernameOrEmail) {
 		User user = null;
 		
-		if (body.containsKey("email")) {
-			String email = body.get("email");
-			user = userService.getUserByEmail(email);
-		}
-		else if (body.containsKey("username")) {
-			String userName = body.get("username");
-			user = userService.getUserByUsername(userName);
+		if ((user = userService.getUserByEmail(usernameOrEmail)) == null) {
+			user = userService.getUserByUsername(usernameOrEmail);
```

Comment:
I think in openmrs-core, you will need to rename this method to `getUserByUsernameOrEmail` just for clarity of what it does